.github/workflows/test_optional_dependencies.yml

```diff
@@ -0,0 +1,23 @@
+name: test_optional_requirements
+
+on: [push, pull_request]
+
+jobs:
+    test_install:
+        strategy:
+            matrix:
+                os: [ "windows-latest", "ubuntu-latest" ] #, "macos-latest"]
+                python-version: [ "3.8", "3.9", "3.10", "3.11" ]
+                extra_requirements: [ "test", "examples", "docs", "dev", "all" ]
+        runs-on: ${{matrix.os}}
+        steps:
+            - uses: actions/checkout@v4
+            # Pull the cache based on the hash value
+            - name: Set up Python ${{ matrix.python-version }}
+              uses: actions/setup-python@v5
+              with:
+                  python-version: ${{ matrix.python-version }}
+                  cache: 'pip'
+            - name: Install dependencies
+              run: |
+                  pip install '.[${{ matrix.extra_requirements }}]'
```

Comment:
Perhaps we can reduce the number of runs by just running "all" for all configurations.